Crash when uploading corrupt or wrong format file into editor

### Expected Behavior

A message should show saying the file couldn't be opened.

### Actual Behavior

Crash - error message shows

### Steps to Reproduce

Click File > Upload form computer
Pick "All files" or type "*.*" in the input box
Pick a non Scratch file, i.e image or word document

### Operating System and Browser

ChomeOS

### Console dumps(if that helps)
raven.js:55 Error: Failed to load project: SyntaxError: Unexpected token � in JSON at position 0
    at t.value (lib.min.js:148)
    at u (lib.min.js:98)
    at beginWork (lib.min.js:98)
    at o (lib.min.js:98)
    at i (lib.min.js:98)
    at a (lib.min.js:98)
    at B (lib.min.js:98)
    at v (lib.min.js:98)
    at f (lib.min.js:98)
    at h (lib.min.js:98)
    at Object.enqueueSetState (lib.min.js:98)
    at t.o.setState (lib.min.js:98)
    at lib.min.js:148
a.(anonymous function) @ raven.js:55
u @ lib.min.js:98
a @ lib.min.js:98
B @ lib.min.js:98
v @ lib.min.js:98
f @ lib.min.js:98
h @ lib.min.js:98
enqueueSetState @ lib.min.js:98
o.setState @ lib.min.js:98
(anonymous) @ lib.min.js:148
Promise.catch (async)
n.onload @ lib.min.js:148
load (async)
value @ lib.min.js:148
l @ lib.min.js:1
u @ lib.min.js:98
invokeGuardedCallback @ lib.min.js:106
invokeGuardedCallbackAndCatchFirstError @ lib.min.js:106
f @ lib.min.js:98
m @ lib.min.js:98
b @ lib.min.js:98
A @ lib.min.js:98
_ @ lib.min.js:98
jl @ lib.min.js:106
Se @ lib.min.js:98
batchedUpdates @ lib.min.js:98
$ @ lib.min.js:98
xe @ lib.min.js:98
d @ raven.js:425
raven.js:55 gui Unhandled Error: Error: Failed to load project: SyntaxError: Unexpected token � in JSON at position 0
    at t.value (https://preview.scratch.mit.edu/lib.min.js:148:40135)
    at u (https://preview.scratch.mit.edu/lib.min.js:98:30488)
    at beginWork (https://preview.scratch.mit.edu/lib.min.js:98:32206)
    at o (https://preview.scratch.mit.edu/lib.min.js:98:44327)
    at i (https://preview.scratch.mit.edu/lib.min.js:98:44648)
    at a (https://preview.scratch.mit.edu/lib.min.js:98:44886)
    at B (https://preview.scratch.mit.edu/lib.min.js:98:48429)
    at v (https://preview.scratch.mit.edu/lib.min.js:98:48166)
    at f (https://preview.scratch.mit.edu/lib.min.js:98:47294)
    at h (https://preview.scratch.mit.edu/lib.min.js:98:46684)
    at Object.enqueueSetState (https://preview.scratch.mit.edu/lib.min.js:98:21265)
    at t.o.setState (https://preview.scratch.mit.edu/lib.min.js:98:568)
    at https://preview.scratch.mit.edu/lib.min.js:148:39645
Component stack: 
    in t
    in Connect(t)
    in A
    in ul
    in h
    in t
    in W
    in div
    in div
    in div
    in div
    in Unknown
    in J
    in Connect(J)
    in div
    in Unknown
    in ue
    in InjectIntl(ue)
    in t
    in Connect(t)
    in n
    in Connect(n)
    in n
    in t
    in IntlProvider
    in Connect(IntlProvider)
    in n
    in Unknown
a.(anonymous function) @ raven.js:55
i.write @ lib.min.js:111
t @ lib.min.js:24
emit @ lib.min.js:111
l.write @ lib.min.js:111
t @ lib.min.js:24
emit @ lib.min.js:111
l.write @ lib.min.js:24
n.error @ lib.min.js:111
value @ lib.min.js:132
t @ lib.min.js:98
B @ lib.min.js:98
v @ lib.min.js:98
f @ lib.min.js:98
h @ lib.min.js:98
enqueueSetState @ lib.min.js:98
o.setState @ lib.min.js:98
(anonymous) @ lib.min.js:148
Promise.catch (async)
n.onload @ lib.min.js:148
load (async)
value @ lib.min.js:148
l @ lib.min.js:1
u @ lib.min.js:98
invokeGuardedCallback @ lib.min.js:106
invokeGuardedCallbackAndCatchFirstError @ lib.min.js:106
f @ lib.min.js:98
m @ lib.min.js:98
b @ lib.min.js:98
A @ lib.min.js:98
_ @ lib.min.js:98
jl @ lib.min.js:106
Se @ lib.min.js:98
batchedUpdates @ lib.min.js:98
$ @ lib.min.js:98
xe @ lib.min.js:98
d @ raven.js:425

--- a/src/containers/project-loader.jsx
+++ b/src/containers/project-loader.jsx
@@ -2,8 +2,10 @@ import bindAll from 'lodash.bindall';
 import PropTypes from 'prop-types';
 import React from 'react';
 import {connect} from 'react-redux';
+import {defineMessages, injectIntl, intlShape} from 'react-intl';
 
 import analytics from '../lib/analytics';
+import log from '../lib/log';
 
 import {
     openLoadingProject,
@@ -26,6 +28,15 @@ import {
  *     </MyCoolComponent>
  * )}</ProjectLoader>
  */
+
+const messages = defineMessages({
+    loadError: {
+        id: 'gui.projectLoader.loadError',
+        defaultMessage: 'The project file that was selected failed to load.',
+        description: 'An error that displays when a local project file fails to load.'
+    }
+});
+
 class ProjectLoader extends React.Component {
     constructor (props) {
         super(props);
@@ -35,10 +46,6 @@ class ProjectLoader extends React.Component {
             'handleChange',
             'handleClick'
         ]);
-        this.state = {
-            loadingError: false,
-            errorMessage: ''
-        };
     }
     handleChange (e) {
         // Remove the hash if any (without triggering a hash change event or a reload)
@@ -52,13 +59,16 @@ class ProjectLoader extends React.Component {
                     action: 'Import Project File',
                     nonInteraction: true
                 });
+            })
+            .catch(error => {
+                log.warn(error);
+                alert(this.props.intl.formatMessage(messages.loadError)); // eslint-disable-line no-alert
+            })
+            .finally(() => {
                 this.props.closeLoadingState();
                 // Reset the file input after project is loaded
                 // This is necessary in case the user wants to reload a project
                 thisFileInput.value = null;
-            })
-            .catch(error => {
-                this.setState({loadingError: true, errorMessage: error});
             });
         if (thisFileInput.files) { // Don't attempt to load if no file was selected
             this.props.openLoadingState();
@@ -83,10 +93,6 @@ class ProjectLoader extends React.Component {
         );
     }
     render () {
-        if (this.state.loadingError) {
-            throw new Error(
-                `Failed to load project from file: ${this.state.errorMessage}`);
-        }
         const {
             /* eslint-disable no-unused-vars */
             children,
@@ -103,6 +109,7 @@ class ProjectLoader extends React.Component {
 ProjectLoader.propTypes = {
     children: PropTypes.func,
     closeLoadingState: PropTypes.func,
+    intl: intlShape.isRequired,
     openLoadingState: PropTypes.func,
     vm: PropTypes.shape({
         loadProject: PropTypes.func
@@ -121,4 +128,4 @@ const mapDispatchToProps = dispatch => ({
 export default connect(
     mapStateToProps,
     mapDispatchToProps
-)(ProjectLoader);
+)(injectIntl(ProjectLoader));

